dashboard/src/components/InteractiveChart.tsx

```diff
@@ -46,7 +46,9 @@ const InteractiveChart: React.FC<InteractiveChartProps> = React.memo(
                   tickLine={false}
                   axisLine={false}
                   className='text-muted-foreground'
-                  tickFormatter={axisFormatter}
+                  tickFormatter={(value) =>
+                    axisFormatter(new Date(typeof value === 'number' ? value : Date.parse(String(value))))
```

Comment:
Nitpick: If the string is already parseable, just skip the parse to number part 😄 
```suggestion
                    axisFormatter(new Date(typeof value === 'number' ? value : String(value)))
```